### DISCUSSION TITLE
[Chapter 08] 상속보다 컴포지션을 이용하는 것이 좋다?

### DISCUSSION THREAD

**Discussion by Irisation23** [cowaine/is-my-code-that-strange#71] [Chapter 07-08 (컬렉션, 강한 결합)] [answered]
8.2.1 슈퍼 클래스 의존으로 인한 강한 결합을 피하려면, 상속보다 컴포지션을 사용하는 것이 좋다는 말이 나옵니다.

-  슈퍼 클래스 의존으로 인한 강한 결합이란 무엇인가요?
- 컴포지션을 사용했을 때 발생할 수 있는 잠재적인 단점은 무엇인가요?

**Comment by firewoody237** [comment, marked as answer]
https://github.com/cowaine/is-my-code-that-strange/discussions/64
위에 강결합 예시를 남겨주셔서 링크드립니다.

단점은 딱히 없어보이지만, 객체를 관리(주입)해야 하거나 상속에비해 설계가 조금 더 복잡하다 정도일 것 같습니다.

**Comment by Irisation23** [comment]
https://hackmd.io/@KangMoo/H1sMlpHZn
